messaging/src/main/java/org/axonframework/eventhandling/pooled/Coordinator.java

```diff
@@ -1074,23 +1099,49 @@ private void abortAndScheduleRetry(Exception cause) {
         }
 
         private CompletableFuture<Void> abortWorkPackage(WorkPackage work, Exception cause) {
+            int segmentId = work.segment().getSegmentId();
             return work.abort(cause)
                        .thenRun(() -> {
-                           if (workPackages.remove(work.segment().getSegmentId(), work)) {
+                           if (workPackages.remove(segmentId, work)) {
                                logger.debug("Processor [{}] released claim on {}.", name, work.segment());
                            }
                        })
                        .thenRun(() -> transactionManager.executeInTransaction(
                                () -> {
-                                   tokenStore.releaseClaim(name, work.segment().getSegmentId());
+                                   tokenStore.releaseClaim(name, segmentId);
                                    segmentReleasedAction.accept(work.segment());
                                }
                        ))
+
                        .exceptionally(throwable -> {
                            logger.info("An exception occurred during the abort of work package [{}] on [{}] processor.",
-                                       work.segment().getSegmentId(), name, throwable);
+                                       segmentId, name, throwable);
                            return null;
                        });
         }
+
+        private void advanceReleaseDeadlineUsingRetryBackOff(int segmentId) {
```

Comment:
Naming suggestion: what if we call this method `calculateReleaseDeadlineFor(int segmentId)`?

In essence, we're calculating how long to keep a given `segmentId` released by **this** `Coordinator`.
The fact it's an exponential back off is of course true, but not necessarily required to be part of the signature, I think.

Reply:
Could be if you prefer like that. I named if with `calculate` before, but then I realized we have some side-effects so is not just "calculating" some value, but I'm not certain about one or another name.

Reply:
I am not religious about calculate or advance, but my opinion is stronger about removing the "retry back off" part from the signature, though.

Reply:
Renamed to `advanceReleaseDeadlineFor` :)